### DISCUSSION TITLE
How to check YOLOv8 training progress while it is already training

### DISCUSSION THREAD

**Discussion by LiranBratt2121** [ultralytics/ultralytics#7769] [Q&A]
I am training a yolov8 model on a large data set (36K Images) 
It is running for more than 4000 minutes on my rtx 2060s gpu.
I forgot to turn on the progress logger feature and I am too deep now to restart the training.
Is there any other way to check the progress of the training? 

Do you guys have an estimation of how much more time will it take to train?

![image](https://github.com/ultralytics/ultralytics/assets/95234790/5dbea8ab-57bc-4680-919a-c54be764e1be)

**Comment by pderrenger** [comment]
@LiranBratt2121 during training, YOLOv8 automatically saves progress logs and checkpoints to a run-specific directory within the `runs/train/` folder. Even if you didn't explicitly enable a progress logger, you can still check the progress by examining these files.

To monitor the training progress, you can:

1. **Check the log files**: Navigate to the `runs/train/exp*/` directory (where `exp*` is the experiment folder, which might have a specific name if you set it). Inside, you'll find a `results.csv` file that logs key metrics like loss and mAP after each epoch.

2. **Review TensorBoard logs**: If TensorBoard was enabled (it's on by default), you can launch TensorBoard to visualize the training metrics. Run `tensorboard --logdir runs/train` in your terminal and open the provided link in a web browser.

3. **Inspect saved weights**: Look for the `last.pt` file in the same `runs/train/exp*/weights/` directory. This file is updated after each epoch, so its timestamp can give you an idea of the last completed epoch.

As for the training time estimation, it's challenging to provide an accurate prediction without knowing specifics like the model architecture, dataset complexity, and hardware specifications. However, since you're using an RTX 2060s, you can roughly estimate the remaining time based on the time taken for the completed epochs. If you know the number of epochs completed and the total epochs planned, you can calculate the remaining time.

For example, if it took 4000 minutes for 50 epochs, and you're training for a total of 100 epochs, you might expect another 4000 minutes for the remaining 50 epochs. Keep in mind that this is a rough estimate and actual times may vary due to factors like varying image complexity and potential system load changes.

Lastly, the image you've referenced seems to be from a placeholder and not an actual training log. If you have specific logs or checkpoints, they will provide more accurate information for progress tracking and time estimation. 😊👨‍💻
